virttest/utils_time.py

```diff
@@ -0,0 +1,229 @@
+import logging
+import re
+
+from avocado.utils import process
+
+from virttest import error_context
+
+
+def get_host_timezone(params):
+    """
+    Timedatectl is installed with systemd.
+    :param params: Dictionary with the test parameters
+
+    :return: timez_city: the host's timezone in city format, used in Linux os.
+    :return: timez_code: the hosts timezone in utc code format, used in both
+                         Linux and Win.
+    """
+    try:
+        o = process.system_output(params["check_host_tz_cmd"], timeout=240,
+                                  shell=True)
+    except process.CmdError:
+        logging.warn("Can not get host's timezone")
+    tz = re.match(r'^(?:\s+Time zone:\s)(\w+\/\w+|UTC)(?:\s\(\w+,\s)([+|-]'
+                  r'\d{4})\)$', o, re.M | re.I)
+    if tz:
+        timez_city = tz.groups()[0]
+        timez_code = tz.groups()[1]
+    else:
+        timez_city = ""
+        timez_code = ""
+    return (timez_city, timez_code)
+
+
+@error_context.context_aware
+def align_guestz_to_hostz(params, vm, session, zonecode, zonecity):
+    """
+    Provide timezone alignment between host and guest.
+
+    This function get the guest's timezone and compare it with host's
+    timezone, if they are same, return True, or change the guest's
+    timezone to the timezone same with host's utc code.
+
+    :param params: Dictionary with the test parameters.
+    :param vm: Virtual machine for vm.
+    :param session: VM session
+    :param zonecode: Host's timezone in code mode.
+    :param zonecity: Host's timezone in city mode.
+
+    :return: True if guest has same timezone with host.
+    """
+    status = -1
+    o_guest = session.cmd_output(params["check_tz_cmd"])
+    logging.info("check timezone in guest: %s" % o_guest)
+    if params.get("os_type") == "linux":
+        guest_tz = re.match(r'^(?:\s+Time zone:\s)(\w+\/\w+|UTC)(?:\s\(\w+'
+                            r',\s)([+|-]\d{4})\)$', o_guest, re.M | re.I)
+        if guest_tz.groups()[1] == zonecode:
+            return (True, session)
```

Comment:
Why return session? It does not changed from it is passed here.

Reply:
Thanks, timezone updating works only when guest is booting from qemu commands again, so we should return a new session.

Reply:
I don't think you need to return the session, you have the vm recreated, the user can get session of the vm object if they need, here this method have no responsibility to open up a session, suppose that user do not need a session after execute this method, it is easily to ignore the returned session open here, and forget close this session.

Reply:
Thanks. I return this session because I introduced one session in arguments, and if this session be closed in this method, the code depends on that session later would fail. I would try a local session in this method and close it before return.